### PR TITLE
feat(VLazy): port to v3

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Added support for resetting lazy content with **model-value**

## Motivation and Context
Port to v3

## How Has This Been Tested?
unit/visually

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-btn style="position: fixed" @click="model = !model">Reset (model={{ model }})</v-btn>
    <div class="ma-12 px-12" style="padding-top: 150vh !important; padding-bottom: 100vh !important;">
      <v-lazy v-model="model" height="200">
        <v-sheet elevation="2">
          <div class="pa-4">
            Hello text
          </div>
        </v-sheet>
      </v-lazy>
    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({ model: false }),
  }
</script>


```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/core-team

File: packages/vuetify/src/components/VLazy/VLazy.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+export default defineComponent({
+  name: 'VLazy',
+
+  directives: { intersect },
+
+  props: makeProps({
+    modelValue: Boolean,
+    options: {
+      type: Object,
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VLazy/VLazy.tsx:22]:
```suggestion
      type: Object as PropType<IntersectionObserverInit>,
```
(i believe it's the correct type)

File: packages/vuetify/src/components/VLazy/VLazy.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    function onIntersect (isIntersecting: boolean) {
+      if (isActive.value) return
+
+      isActive.value = isIntersecting
+    }
+
+    return () => (
+      <props.tag
+        class="v-lazy"
+        v-intersect={[
```
Comment (nekosaur) [inline packages/vuetify/src/components/VLazy/VLazy.tsx:53]: use `useDirective`

File: packages/vuetify/src/components/VLazy/VLazy.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          onIntersect,
+          props.options,
+          ['once'],
+        ]}
+        style={dimensionStyles.value}
+      >
+        { isActive.value && (
+          maybeTransition(
```
Comment (nekosaur) [inline packages/vuetify/src/components/VLazy/VLazy.tsx:61]: I thought @KaelWD made this a component, but now I can't find it?

File: packages/vuetify/src/components/VLazy/VLazy.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
+// Composables
+import { makeDimensionProps, useDimension } from '@/composables/dimensions'
+import { makeProps, maybeTransition } from '@/util'
```
Comment (nekosaur) [inline packages/vuetify/src/components/VLazy/VLazy.tsx:3]: not composables

**Review by nekosaur — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/components/VLazy/VLazy.tsx (hunk 3 above)
Comment (nekosaur) [reply packages/vuetify/src/components/VLazy/VLazy.tsx:61]: Nevermind that was in his overlay branch

File: packages/vuetify/src/components/VLazy/VLazy.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      isActive.value = isIntersecting
+    }
+
+    return () => (
+      <props.tag
+        class="v-lazy"
+        v-intersect={[
+          onIntersect,
+          props.options,
+          ['once'],
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VLazy/VLazy.tsx:56]: if it really is intended to work only once then `modelValue` doesn't make much sense imho

```vue
<template>
  <v-app>
    <v-btn style="position: fixed" @click="model = false">Reset (model={{ model }})</v-btn>
    <div class="ma-12 px-12" style="padding-top: 150vh !important; padding-bottom: 100vh !important;">
      <v-lazy v-model="model" height="200">
        <v-sheet elevation="2">
          <div class="pa-4">
            Hello text
          </div>
        </v-sheet>
      </v-lazy>
    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({ model: false }),
  }
</script>
```

File: packages/vuetify/src/components/VLazy/VLazy.tsx (hunk 2 above)
Comment (nekosaur) [reply packages/vuetify/src/components/VLazy/VLazy.tsx:53]: nm, looks like we're using directives instead?

Comment (johnleider) [reply packages/vuetify/src/components/VLazy/VLazy.tsx:53]: yes

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/core-team

**Pull request merged**